src/core/drive/page_snapshot.ts

```diff
@@ -1,10 +1,20 @@
+import { CSPTrustedTypesPolicy, emptyHTML } from "../../trusted_types"
 import { parseHTMLDocument } from "../../util"
 import { Snapshot } from "../snapshot"
 import { expandURL } from "../url"
 import { HeadSnapshot } from "./head_snapshot"
 
 export class PageSnapshot extends Snapshot<HTMLBodyElement> {
-  static fromHTMLString(html = "") {
+  static fromResponse(response?: Response, html = emptyHTML): PageSnapshot {
+    if (!response || CSPTrustedTypesPolicy == null) {
```

Comment:
out of curiosity, when do we expect there to be no response?

Reply:
https://github.com/github/turbo/blob/b013ad76c317515e6b4a6cbc75673d757902e376/src/core/drive/visit.ts#L267

In this case it's possible for the responseHTML to be null